torch/onnx/utils.py

```diff
@@ -141,6 +141,8 @@ def _optimize_graph(graph, operator_export_type):
     torch._C._jit_pass_peephole(graph)
     torch._C._jit_pass_lint(graph)
 
+    # onnx only supports tensors, but 1 / 2 = 0.5 and tensor(1) / tensor(2) = 0
+    torch._C._jit_pass_prepare_division_for_onnx(graph)
```

Comment:
There is a RAW mode of ONNX export where everything gets turned into ATen-in-caffe2 operators (see the if statement below), but this pass will introduce onnx::Cast operators even in that mode. Is it not possible to do this fixup as part of `def div` in symbolic.py?

Reply:
Let me investigate and get back to you

Reply:
There isn't an easy way to do the fixup as part of the `def div` in symbolic.py -- EraseNumberTypes runs before `div` in symbolic gets called. Because of this, `div` doesn't have an easy way to check if it is dividing two numbers or two scalar tensors (it sees two scalar tensors).  The way to distinguish between the two is check the type of the outputs in the graph (LongTensor, LongTensor -> FloatTensor) but `div` doesn't get that information.

I ended up following @jamesr66a's advice and used `aten::_cast_Float` instead of `onnx::Cast` (thanks James!), let me know if this looks good.